src/libPMacc/include/math/MapTuple.hpp

```diff
@@ -23,199 +23,215 @@
-class MapTuple;
+        HDINLINE AlignedData()
+        {
+        }
 
-template<typename Map_, template<typename> class PODType>
-class MapTuple<Map_, PODType, true>
-{
-};
+        HDINLINE AlignedData(const ValueType& value) : value(value)
+        {
+        }
 
-template<typename Map_, template<typename> class PODType>
-class MapTuple<Map_, PODType, false>
-: public MapTuple<typename mpl::erase_key<
-Map_, typename mpl::deref<typename mpl::begin<Map_>::type>::type::first>::type, PODType
->
-{
-public:
-    typedef Map_ Map;
-    BOOST_STATIC_CONSTEXPR int dim = mpl::size<Map>::type::value;
-private:
-   
-    typedef MapTuple<typename mpl::erase_key<
-    Map, typename mpl::deref<typename mpl::begin<Map>::type>::type::first>::type, PODType> base;
+        HDINLINE ValueType& operator[](const Key&)
+        {
+            return value;
+        }
 
-    typedef typename mpl::deref<typename mpl::begin<Map>::type>::type::first Key;
-    typedef typename mpl::deref<typename mpl::begin<Map>::type>::type::second Value;
-
-    PODType<Value> data;
-public:
-
-    template<class> struct result;
-
-    template<class F, class TKey>
-    struct result<F(TKey)>
-    {
-        typedef typename mpl::at<Map, TKey>::type& type;
-       // typedef typename mpl::at<F, TKey>::type& type2;
+        HDINLINE const ValueType& operator[](const Key&) const
+        {
+            return value;
+        }
     };
 
-    template<class F, class TKey>
-    struct result<const F(TKey)>
+    /** wrap a datum
+     *
+     * @tparam T_Pair boost mpl pair< key, type of the value >
+     */
+    template< typename T_Pair >
+    struct NativeData
     {
+        typedef typename T_Pair::first Key;
+        typedef typename T_Pair::second ValueType;
 
-        typedef const typename mpl::at<Map, TKey>::type& type;
-    };
-/*
-    template<class T_Map,template<typename> class T_PODType,bool T_isEnd,class TKey>
-    struct result<const MapTuple<T_Map,T_PODType,T_isEnd>(TKey)>
-    {
-        typedef const typename mpl::at<Map, TKey>::type& type;
-    };
-*/
-    HDINLINE MapTuple()
-    {
-    }
+        ValueType value;
 
-    template<typename Arg0>
-    HDINLINE MapTuple(const Arg0& arg0) : data(arg0)
-    {
-        BOOST_STATIC_ASSERT(dim == 1);
-    }
+        HDINLINE NativeData()
+        {
+        }
 
-    BOOST_PP_REPEAT_FROM_TO(2, BOOST_PP_INC(MAPTUPLE_MAX_DIM), CONSTRUCTOR, _)
+        HDINLINE NativeData(const ValueType& value) : value(value)
+        {
+        }
 
+        HDINLINE ValueType& operator[](const Key&)
+        {
+            return value;
+        }
 
-    HDINLINE Value& operator[](const Key)
-    {
-        return this->data.value;
-    }
+        HDINLINE const ValueType& operator[](const Key&) const
+        {
+            return value;
+        }
+    };
 
-    HDINLINE const Value& operator[](const Key) const
+    template<typename Map_, template<typename> class PODType = NativeData>
```

Comment:
Yes you are right I leave this part as it is to minimize the changes. But This class is fully refactored therefore I can change the names too.